### ISSUE TITLE
I am confused with config server,  transporter config.

### ISSUE THREAD

**Issue by whanice** [apache/incubator-dubbo#1841]
In my  understanding.
if we config  transporter="netty4" in provider side. It will  affect the consumer。
so we can use  server="netty4" in provider side.

but why dubbo  check sever key in `DubboProtocol.initClient` ?
the consumer with low version (not support netty4) will start failed.

And the transporter key will not  affect the consumer now.

**Comment by carryxyh** [comment]
transport config will not override.
the config that can override is like timeout, retries, loadbalance, actives and so on.

**Comment by zonghaishang** [comment]
@whanice
1. `if we config transporter="netty4" in provider side. It will affect the consumer。`
Transporter is the protocolconfig attribute， the client does not have a protocolconfig concept, therefore the tranporter on the provider side will not be passed to the client.
2. `but why dubbo check sever key in DubboProtocol.initClient ?`
Because the value stored by the server will be passed to the client,therefore, the client will try to take the value of the server first.
3. `the consumer with low version (not support netty4) will start failed.`
I have tried neety4 as a provider, netty3 as a client, and can use it normally, Convenient to provide demo program verification?

Thank you very much for your support of dubbo.

**Comment by whanice** [comment]
Thanks for your guys reply.
Surely. the tcp connection should not be related to the nio framework,  No matter what you use, netty3, netty4 or mina in server. Will not affect the client use. 

I care about migration.   when consumer use dubbo-2.5.x .  not have netty4 extension.
when provider  wants to upgrade dubbo verison to dubbo-2.6.x and use netty4. It will affect the consumer.  

So I think in `DubboProtol.initClient`.  change to  will be better.
```
 // client type setting.
    String str = url.getParameter(Constants.CLIENT_KEY, url.getParameter(Constants.TRANSPORTER_KEY, Constants.DEFAULT_REMOTING_CLIENT));
```

**Comment by whanice** [comment]
And in the future. I think the sever ket shoud not pass to consumer.

**Comment by zonghaishang** [comment]
You are right, I will confirm this question again.

**Comment by zonghaishang** [comment]
Through communication, we keep netty as the extension point name, there will be no problem

**Comment by chickenlj** [comment]
> And in the future. I think the sever key shoud not pass to consumer.

Agree, and this can be solved with #2030
